### ISSUE TITLE
UserWarning:  This call to matplotlib.use() has no effect

### ISSUE THREAD

**Issue by yoavram** [ppinard/matplotlib-colorbar#9]
When importing `from matplotlib_colorbar.colorbar import Colorbar` after matplotlib has already been imported, say with `%matplotlib inline`, a warning comes up:
```
/Users/username/anaconda3/envs/scipy/lib/python3.6/site-packages/ipykernel/__main__.py:8: UserWarning: 
This call to matplotlib.use() has no effect because the backend has already
been chosen; matplotlib.use() must be called *before* pylab, matplotlib.pyplot,
or matplotlib.backends is imported for the first time.
```

I managed to [trace](https://stackoverflow.com/a/29567225/1063612) the warning, and it's coming from `matplotlib_colorbar/colorbar.py`, line 83:
```py
imp.reload(sys.modules['matplotlib'])
```

If this is OK, maybe you can suppress the warning?

```py
import warnings
with warnings.catch_warnings():
    warnings.simplefilter("ignore", UserWarning)
    imp.reload(sys.modules['matplotlib'])
```

**Comment by ppinard** [comment]
Thank you for reporting this bug. I saw the same warning message in my other package matplotlib-scalebar. I applied the same fix to matplotlib-colorbar which consists in reinitializing only the validate function.

```
matplotlib.rcParams.validate = \
    dict((key, converter) for key, (default, converter) in
         six.iteritems(defaultParams)
         if key not in matplotlib._all_deprecated)
```

It seems to work on my system, but I would appreciate if you have time to check it on yours. If it works, I will release a new version.
